### PR TITLE
Functions refactor synced with 0.13.x

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

Sync 0.13.x with Functions refactor

## Test Plan

Existing tests

## Related PRs and Issues

https://github.com/appwrite/appwrite/pull/2810

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

File: app/config/collections.php [hunk 1]
```diff
@@ -1932,9 +1751,9 @@
                 '$id' => '_key_search',
                 'type' => Database::INDEX_FULLTEXT,
                 'attributes' => ['search'],
-                'lengths' => [2048],
-                'orders' => [Database::ORDER_ASC],
-            ]
+                'lengths' => [],
+                'orders' => [],
+            ],
```
Comment (christyjacob4) [inline app/config/collections.php:1756]: Can you revert the change to the functions collection. This change shouldn't be a part of this PR

File: app/config/variables.php [hunk 2]
```diff
@@ -420,12 +420,93 @@
...
+            [
+                'name' => '_APP_STORAGE_DEVICE',
+                'description' => 'Select default storage device. The default value is \'Local\'. List of supported adapters are \'Local\', \'S3\' and \'DOSpaces\'.',
+                'introduction' => '0.13.0',
+                'default' => 'Local',
+                'required' => false,
+                'question' => '',
+            ],
+            [
+                'name' => '_APP_STORAGE_DEVICE_S3_ACCESS_KEY',
```
Comment (christyjacob4) [inline app/config/variables.php:432]:
```suggestion
                'name' => '_APP_STORAGE_S3_ACCESS_KEY',
```
The DEVICE word seems unnecessary as it does not add any additional meaning to the already existing word `_APP_STORAGE_S3_ACCESS_KEY` . I recommend discarding the DEVICE word and keeping the key name shorter.

File: app/config/variables.php [hunk 3]
```diff
@@ -420,12 +420,93 @@
...
+                'description' => 'AWS S3 storage access key. Required when the storage adapter is set to S3. You can get your access key from your AWS console',
+                'introduction' => '0.13.0',
+                'default' => '',
+                'required' => false,
+                'question' => '',
+            ],
+            [
+                'name' => '_APP_STORAGE_DEVICE_S3_SECRET',
```
Comment (christyjacob4) [inline app/config/variables.php:440]:
```suggestion
                'name' => '_APP_STORAGE_S3_SECRET',
```
Similar comment as above

File: app/config/variables.php [hunk 4]
```diff
@@ -420,12 +420,93 @@
...
+                'description' => 'AWS S3 storage secret key. Required when the storage adapter is set to S3. You can get your secret key from your AWS console.',
+                'introduction' => '0.13.0',
+                'default' => '',
+                'required' => false,
+                'question' => '',
+            ],
+            [
+                'name' => '_APP_STORAGE_DEVICE_S3_REGION',
```
Comment (christyjacob4) [inline app/config/variables.php:448]:
```suggestion
                'name' => '_APP_STORAGE_S3_REGION',
```

File: app/config/variables.php [hunk 5]
```diff
@@ -420,12 +420,93 @@
...
+                'description' => 'AWS S3 storage region. Required when storage adapter is set to S3. You can find your region info for your bucket from AWS console.',
+                'introduction' => '0.13.0',
+                'default' => 'us-eas-1',
+                'required' => false,
+                'question' => '',
+            ],
+            [
+                'name' => '_APP_STORAGE_DEVICE_S3_BUCKET',
```
Comment (christyjacob4) [inline app/config/variables.php:456]:
```suggestion
                'name' => '_APP_STORAGE_S3_BUCKET',
```

File: app/config/variables.php [hunk 6]
```diff
@@ -420,12 +420,93 @@
...
+                'description' => 'AWS S3 storage bucket. Required when storage adapter is set to S3. You can create buckets in your AWS console.',
+                'introduction' => '0.13.0',
+                'default' => '',
+                'required' => false,
+                'question' => '',
+            ],
+            [
+                'name' => '_APP_STORAGE_DEVICE_DO_SPACES_ACCESS_KEY',
```
Comment (christyjacob4) [inline app/config/variables.php:464]:
```suggestion
                'name' => '_APP_STORAGE_DO_SPACES_ACCESS_KEY',
```

File: app/config/variables.php [hunk 7]
```diff
@@ -420,12 +420,93 @@
...
+                'description' => 'DigitalOcean spaces access key. Required when the storage adapter is set to DOSpaces. You can get your access key from your DigitalOcean console.',
+                'introduction' => '0.13.0',
+                'default' => '',
+                'required' => false,
+                'question' => '',
+            ],
+            [
+                'name' => '_APP_STORAGE_DEVICE_DO_SPACES_SECRET',
```
Comment (christyjacob4) [inline app/config/variables.php:472]:
```suggestion
                'name' => '_APP_STORAGE_DO_SPACES_SECRET',
```

File: app/config/variables.php [hunk 8]
```diff
@@ -420,12 +420,93 @@
...
+                'description' => 'DigitalOcean spaces secret key. Required when the storage adapter is set to DOSpaces. You can get your secret key from your DigitalOcean console.',
+                'introduction' => '0.13.0',
+                'default' => '',
+                'required' => false,
+                'question' => '',
+            ],
+            [
+                'name' => '_APP_STORAGE_DEVICE_DO_SPACES_REGION',
```
Comment (christyjacob4) [inline app/config/variables.php:480]:
```suggestion
                'name' => '_APP_STORAGE_DO_SPACES_REGION',
```

File: app/config/variables.php [hunk 9]
```diff
@@ -420,12 +420,93 @@
...
+                'description' => 'DigitalOcean spaces region. Required when storage adapter is set to DOSpaces. You can find your region info for your space from DigitalOcean console.',
+                'introduction' => '0.13.0',
+                'default' => 'us-eas-1',
+                'required' => false,
+                'question' => '',
+            ],
+            [
+                'name' => '_APP_STORAGE_DEVICE_DO_SPACES_BUCKET',
```
Comment (christyjacob4) [inline app/config/variables.php:488]:
```suggestion
                'name' => '_APP_STORAGE_DO_SPACES_BUCKET',
```

File: app/config/variables.php [hunk 10]
```diff
@@ -420,12 +420,93 @@
...
+                'question' => '',
+            ],
         ],
     ],
     [
         'category' => 'Functions',
         'description' => '',
         'variables' => [
+            [
+                'name' => '_APP_FUNCTIONS_DEPLOYMENT_LIMIT',
```
Comment (christyjacob4) [inline app/config/variables.php:502]:
```suggestion
                'name' => '_APP_FUNCTIONS_SIZE_LIMIT',
```

I feel `_APP_FUNCTIONS_SIZE_LIMIT` or `_APP_DEPLOYMENT_SIZE_LIMIT` conveys the meaning better.  LMK your thoughts

File: app/config/collections.php (hunk 1 above)
Comment (lohanidamodar) [reply app/config/collections.php:1756]: what do you mean?

File: app/config/collections.php [hunk 11]
```diff
@@ -2021,6 +1840,50 @@
...
+                '$id' => 'metadata',
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => 16384, // https://tools.ietf.org/html/rfc4288#section-4.2
+                'signed' => true,
+                'required' => false,
+                'default' => null,
+                'array' => false,
+                'filters' => ['json'],
+            ],
```
Comment (christyjacob4) [inline app/config/collections.php:1886]:
```suggestion
```
Looks like metadata is declared twice

File: app/config/variables.php (hunk 10 above)
Comment (lohanidamodar) [reply app/config/variables.php:502]: _APP_FUNCTIONS because it's related to functions and in functions category, it's not a function's size limit but limit for DEPLOYMENT_SIZE. WDYT?

File: app/controllers/api/functions.php [hunk 12]
```diff
@@ -535,33 +574,62 @@
...
-            'type' => BUILD_TYPE_DEPLOYMENT
-        ]);
+        if($chunksUploaded === $chunks) {
+            $fileSize = $deviceFunctions->getFileSize($path);
+
+            if ($deployment->isEmpty()) {
+                $deployment = $dbForProject->createDocument('deployments', new Document([
+                    '$id' => $deploymentId,
+                    '$read' => ['role:all'],
+                    '$write' => ['role:all'],
```
Comment (christyjacob4) [inline app/controllers/api/functions.php:584]:
```suggestion
                    '$read' => [],
                    '$write' => [],
```
The `role:all` part does not seem secure.. @TorstenDittmann  @eldadfux  should we just leave the permissions empty ?

File: app/controllers/api/functions.php [hunk 13]
```diff
@@ -535,33 +574,62 @@
...
+                    'resourceId' => $function->getId(),
+                    'resourceType' => 'functions',
+                    'dateCreated' => time(),
+                    'entrypoint' => $entrypoint,
+                    'path' => $path,
+                    'size' => $fileSize,
+                    'search' => implode(' ', [$deploymentId, $entrypoint]),
+                    'activate' => ((bool) $activate === true),
```
Comment (christyjacob4) [inline app/controllers/api/functions.php:592]:
```suggestion
                    'activate' => $activate,
```

File: app/controllers/api/functions.php [hunk 14]
```diff
@@ -535,33 +574,62 @@
...
+                    'resourceId' => $function->getId(),
+                    'resourceType' => 'functions',
+                    'dateCreated' => time(),
+                    'entrypoint' => $entrypoint,
+                    'path' => $path,
+                    'size' => 0,
+                    'chunksTotal' => $chunks,
+                    'chunksUploaded' => $chunksUploaded,
+                    'search' => implode(' ', [$deploymentId, $entrypoint]),
+                    'activate' => ((bool) $activate === true),
```
Comment (christyjacob4) [inline app/controllers/api/functions.php:624]:
```suggestion
                    'activate' => $activate,
```

File: app/controllers/api/functions.php [hunk 15]
```diff
@@ -698,15 +769,17 @@
...
         }
 
         if ($deployment->getAttribute('resourceId') !== $function->getId()) {
-            throw new Exception('Deployment not found', 404);
+            throw new Exception('Deployment not found', 404, Exception::DEPLOYMENT_NOT_FOUND);
         }
 
-        if (!$dbForProject->deleteDocument('deployments', $deployment->getId())) {
-            throw new Exception('Failed to remove deployment from DB', 500);
+        if ($deviceFunctions->delete($deployment->getAttribute('path', ''))) {
```
Comment (christyjacob4) [inline app/controllers/api/functions.php:779]: We shouldn't be doing any delete operations here. The deletes worker should be doing it in the background

File: app/controllers/api/functions.php [hunk 16]
```diff
@@ -535,33 +574,62 @@
...
-        ;
+            $usage
+                ->setParam('storage', $deployment->getAttribute('size', 0))
+            ;
+        } else {
+            if($deployment->isEmpty()) {
+                $deployment = $dbForProject->createDocument('deployments', new Document([
+                    '$id' => $deploymentId,
+                    '$read' => ['role:all'],
+                    '$write' => ['role:all'],
```
Comment (christyjacob4) [inline app/controllers/api/functions.php:614]: similar comments about the permissions

File: app/controllers/api/functions.php (hunk 12 above)
Comment (lohanidamodar) [reply app/controllers/api/functions.php:584]: @christyjacob4 took from your branch?
https://github.com/appwrite/appwrite/blob/e9abe0f943b9e932bdb990fa594a9d555f5ebc4a/app/controllers/api/functions.php#L541

File: app/controllers/api/functions.php (hunk 15 above)
Comment (lohanidamodar) [reply app/controllers/api/functions.php:779]: is this handled in the background?

File: app/controllers/api/functions.php [hunk 17]
```diff
@@ -535,33 +574,62 @@
...
+                    'resourceId' => $function->getId(),
+                    'resourceType' => 'functions',
+                    'dateCreated' => time(),
+                    'entrypoint' => $entrypoint,
+                    'path' => $path,
+                    'size' => 0,
```
[the remark this answers is not in the document]
Comment (lohanidamodar) [reply app/controllers/api/functions.php]: fixed

File: app/controllers/api/functions.php (hunk 16 above)
Comment (lohanidamodar) [reply app/controllers/api/functions.php:614]: you have the same in your branch?

File: app/controllers/api/storage.php [hunk 18]
```diff
@@ -2,32 +2,338 @@
...
+    ->label('event', 'storage.buckets.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'createBucket')
+    ->label('sdk.description', '/docs/references/storage/create-bucket.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_BUCKET)
+    ->param('bucketId', '', new CustomId(), 'Unique Id. Choose your own unique ID or pass the string `unique()` to auto generate it. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
+    ->param('name', '', new Text(128), 'Bucket name', false)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:51]:
```suggestion
    ->param('name', '', new Text(128), 'Bucket name')
```
the `optional` param is `false` by default. might be a good idea to skip it

File: app/controllers/api/storage.php [hunk 19]
```diff
@@ -2,32 +2,338 @@
...
+    ->param('enabled', true, new Boolean(), 'Is bucket enabled?', true)
+    ->param('maximumFileSize', (int) App::getEnv('_APP_STORAGE_LIMIT', 0), new Integer(), 'Maximum file size allowed in bytes. Maximum allowed value is ' . Storage::human(App::getEnv('_APP_STORAGE_LIMIT', 0), 0) . '. For self-hosted setups you can change the max limit by changing the `_APP_STORAGE_LIMIT` environment variable. [Learn more about storage environment variables](docs/environment-variables#storage)', true)
+    ->param('allowedFileExtensions', [], new ArrayList(new Text(64)), 'Allowed file extensions', true)
+    ->param('encryption', true, new Boolean(), 'Is encryption enabled? For file size above ' . Storage::human(APP_STORAGE_READ_BUFFER, 0) . ' encryption is skipped even if it\'s enabled', true)
+    ->param('antivirus', true, new Boolean(), 'Is virus scanning enabled? For file size above ' . Storage::human(APP_LIMIT_ANTIVIRUS, 0) . ' AntiVirus scanning is skipped even if it\'s enabled', true)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:59]: When using the Boolean validator, I think it's important to use the `loose` argument to true.
Since some SDKs ( like JS ) send booleans as strings. and without the loose argument, it would fail for a valid string like "true" or "false". 

Also when dealing with booleans represented by strings, it's a good idea to use PHPs `filter_var` function to cast it to a boolean like we do in functions 
https://github.com/appwrite/appwrite/blob/feat-functions-refactor/app/controllers/api/functions.php#L522

File: app/controllers/api/storage.php [hunk 20]
```diff
@@ -2,32 +2,338 @@
...
+    ->param('name', '', new Text(128), 'Bucket name', false)
+    ->param('permission', null, new WhiteList(['file', 'bucket']), 'Permissions type model to use for reading files in this bucket. You can use bucket-level permission set once on the bucket using the `read` and `write` params, or you can set file-level permission where each file read and write params will decide who has access to read and write to each file individually. [learn more about permissions](/docs/permissions) and get a full list of available permissions.')
+    ->param('read', null, new Permissions(), 'An array of strings with read permissions. By default no user is granted with any read permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
+    ->param('write', null, new Permissions(), 'An array of strings with write permissions. By default no user is granted with any write permissions. [learn more about permissions](/docs/permissions) and get a full list of available permissions.', true)
+    ->param('enabled', true, new Boolean(), 'Is bucket enabled?', true)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:55]:
```suggestion
    ->param('enabled', true, new Boolean(true), 'Is bucket enabled?', true)
```

File: app/controllers/api/storage.php [hunk 21]
```diff
@@ -2,32 +2,338 @@
...
+    ->inject('usage')
+    ->action(function ($bucketId, $name, $permission, $read, $write, $enabled, $maximumFileSize, $allowedFileExtensions, $encryption, $antivirus, $response, $dbForProject, $audits, $usage) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForProject */
+        /** @var Appwrite\Event\Event $audits */
+        /** @var Appwrite\Stats\Stats $usage */
+
+        $bucketId = $bucketId === 'unique()' ? $dbForProject->getId() : $bucketId;
+        try {
+            $files = Config::getParam('collections', [])['files'] ?? [];
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:72]: I think trying to access `files` without using `isset()` to check it first, causes a warning ( or throws an error ) 🤔

File: public/scripts/views/forms/upload.js [hunk 22]
```diff
@@ -108,7 +108,7 @@
           expression.parse(element.dataset["write"] || "[]")
         );
 
-        sdk.storage.createFile('unique()', files[0], read, write, 1).then(
+        sdk.storage.createFile('default', 'unique()', files[0], read, write, 1).then(
           function(response) {
```
Comment (christyjacob4) [inline public/scripts/views/forms/upload.js]: Any reason we chose to have the bucket as the first param in create file  ? 🤔 
Every other  Appwrite endpoint that creates something, expects the ID first 🤔 .
Even the `createDocument` endpoint. What do you think ?

File: app/init.php [hunk 23]
```diff
@@ -60,6 +66,10 @@
 const APP_PAGING_LIMIT = 12;
 const APP_LIMIT_COUNT = 5000;
 const APP_LIMIT_USERS = 10000;
+const APP_LIMIT_ANTIVIRUS = 20000000; //20MB
+const APP_LIMIT_ENCRYPTION = 20000000; //20MB
+const APP_LIMIT_COMPRESSION = 20000000; //20MB
+const APP_LIMIT_PREVIEW = 10000000; //10MB file size limit for preview endpoint
```
Comment (christyjacob4) [inline app/init.php:72]:
```suggestion
const APP_LIMIT_PREVIEW = 10 * (1000 * 1000); //10MB file size limit for preview endpoint
```

Might be nice to split it like this as you have done for `APP_STORAGE_READ_BUFFER `

File: app/init.php [hunk 24]
```diff
@@ -60,6 +66,10 @@
 const APP_PAGING_LIMIT = 12;
 const APP_LIMIT_COUNT = 5000;
 const APP_LIMIT_USERS = 10000;
+const APP_LIMIT_ANTIVIRUS = 20000000; //20MB
+const APP_LIMIT_ENCRYPTION = 20000000; //20MB
+const APP_LIMIT_COMPRESSION = 20000000; //20MB
```
Comment (christyjacob4) [inline app/init.php:71]: Might be nice to split the number like you did for APP_STORAGE_READ_BUFFER

File: app/controllers/api/storage.php (hunk 21 above)
Comment (lohanidamodar) [reply app/controllers/api/storage.php:72]: the null check operator `?? []` prevents that.

File: public/scripts/views/forms/upload.js (hunk 22 above)
Comment (lohanidamodar) [reply public/scripts/views/forms/upload.js:112]: I was not aware of this change.

File: app/init.php (hunk 23 above)
Comment (lohanidamodar) [reply app/init.php:72]: I'm doing it for APP_STORAGE_READ_BUFFER because it's using MiB (1024), but this is using metric system and with comment //10 MB it's quite obvious, same for other similar vars?

File: app/init.php (hunk 24 above)
Comment (lohanidamodar) [reply app/init.php:71]: answered above

File: src/Appwrite/Utopia/Response/Model/Bucket.php [hunk 25]
```diff
@@ -0,0 +1,111 @@
...
+use Appwrite\Utopia\Response;
+use Appwrite\Utopia\Response\Model;
+
+class Bucket extends Model
+{
+
+    public function __construct()
+    {
+        $this
+        ->addRule('$id', [
```
Comment (christyjacob4) [inline src/Appwrite/Utopia/Response/Model/Bucket.php]: Very minor, looks like we use 1 level of indent here . Would be nice to maintain consistency 

https://github.com/appwrite/appwrite/blob/feat-functions-refactor/src/Appwrite/Utopia/Response/Model/Log.php#L13

File: tests/e2e/Services/Functions/FunctionsCustomServerTest.php [hunk 26]
```diff
@@ -303,6 +299,41 @@ public function testCreateDeployment($data):array
 
         // Wait for deployment to build.
         sleep(15);
+
+        /**
+         * Test for Large Code File SUCCESS
+         */
+        $source = realpath(__DIR__ . '/../../../resources/functions/php-large.tar.gz');
```
Comment (christyjacob4) [inline tests/e2e/Services/Functions/FunctionsCustomServerTest.php]: We have a easier way to do this
https://github.com/appwrite/appwrite/blob/feat-functions-refactor/tests/e2e/Services/Functions/FunctionsCustomServerTest.php#L285-L295

We don't keep tar files in the repo anymore. The test suite creates the tar files from the source code during runtime. 

So you can remove this file as well
tests/resources/functions/php-large.tar.gz

File: tests/e2e/Services/Functions/FunctionsCustomServerTest.php [hunk 27]
```diff
@@ -303,6 +299,41 @@ public function testCreateDeployment($data):array
...
+            $counter++;
+            $id = $largeTag['body']['$id'];
+        }
+        @fclose($handle);
+
+        $this->assertEquals(201, $largeTag['headers']['status-code']);
+        $this->assertNotEmpty($largeTag['body']['$id']);
+        $this->assertIsInt($largeTag['body']['dateCreated']);
+        $this->assertEquals('index.php', $largeTag['body']['entrypoint']);
+        $this->assertGreaterThan(10000, $largeTag['body']['size']);
```
Comment (christyjacob4) [inline tests/e2e/Services/Functions/FunctionsCustomServerTest.php]: Also, let's keep the large file upload test, as an independent test and not couple it with the existing `createDeployment` test. 

Let's do something like the `testCreateCustomNodeExecution()` method . This creates a deployment, an execution and even cleans up the container after the test. 

https://github.com/appwrite/appwrite/blob/feat-functions-refactor/tests/e2e/Services/Functions/FunctionsCustomServerTest.php#L851

File: app/http.php [hunk 28]
```diff
@@ -141,8 +144,61 @@
...
+                'dateUpdated' => \time(),
+                'name' => 'Default',
+                'permission' => 'file',
+                'maximumFileSize' => (int) App::getEnv('_APP_STORAGE_LIMIT', 0), // 10MB
+                'allowedFileExtensions' => [],
+                'enabled' => true,
+                'encryption' => true,
+                'antivirus' => true,
+                '$read' => ['role:all'],
+                '$write' => ['role:all'],
```
Comment (christyjacob4) [inline app/http.php:164]: Are we sure about these permissions ?

File: app/controllers/api/storage.php [hunk 29]
```diff
@@ -2,32 +2,338 @@
...
+    ->param('enabled', true, new Boolean(), 'Is bucket enabled?', true)
+    ->param('maximumFileSize', null, new Integer(), 'Maximum file size allowed in bytes. Maximum allowed value is ' . Storage::human((int)App::getEnv('_APP_STORAGE_LIMIT', 0), 0) . '. For self hosted version you can change the limit by changing _APP_STORAGE_LIMIT environment variable. [Learn more about storage environment variables](docs/environment-variables#storage)', true)
+    ->param('allowedFileExtensions', [], new ArrayList(new Text(64)), 'Allowed file extensions', true)
+    ->param('encryption', true, new Boolean(), 'Is encryption enabled? For file size above ' . Storage::human(APP_STORAGE_READ_BUFFER, 0) . ' encryption is skipped even if it\'s enabled', true)
+    ->param('antivirus', true, new Boolean(), 'Is virus scanning enabled? For file size above ' . Storage::human(APP_LIMIT_ANTIVIRUS, 0) . ' AntiVirus scanning is skipped even if it\'s enabled', true)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:231]: Please refer to the comment above about booleans

File: app/controllers/api/storage.php [hunk 30]
```diff
@@ -41,150 +347,345 @@
...
 
-        /*
-         * Validators
-         */
-        //$fileType = new FileType(array(FileType::FILE_TYPE_PNG, FileType::FILE_TYPE_GIF, FileType::FILE_TYPE_JPEG));
-        $fileSize = new FileSize(App::getEnv('_APP_STORAGE_LIMIT', 0));
-        $upload = new Upload();
+        /**
+     * Validators
+     */
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:416]: Can we get these aligned ?

File: app/controllers/api/storage.php [hunk 31]
```diff
@@ -245,28 +785,53 @@
     ->label('sdk.response.code', Response::STATUS_CODE_OK)
     ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
     ->label('sdk.response.model', Response::MODEL_FILE)
+    ->param('bucketId', null, new UID(), 'Storage bucket unique ID. You can create a new storage bucket using the Storage service [server integration](/docs/server/storage#createBucket).')
     ->param('fileId', '', new UID(), 'File ID.')
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:789]: Don't we have an upper limit on the number of characters for an ID ? Shouldn't we specify it as part of the UID() constructor ?

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

File: app/controllers/api/storage.php (hunk 20 above)
Comment (christyjacob4) [reply app/controllers/api/storage.php:55]: You can refer to the comment above 
https://github.com/appwrite/appwrite/pull/2818#discussion_r811911899

File: public/scripts/views/forms/upload.js (hunk 22 above)
Comment (lohanidamodar) [reply public/scripts/views/forms/upload.js:112]: I don't think that's the case,  as collectionId, bucketId being the path param it comes before body param
https://appwrite.io/docs/client/database?sdk=web-default#databaseCreateDocument

File: app/http.php (hunk 28 above)
Comment (lohanidamodar) [reply app/http.php:164]: yes, as permission is set to file level, this shouldn't matter

File: app/controllers/api/storage.php (hunk 31 above)
Comment (lohanidamodar) [reply app/controllers/api/storage.php:789]: It's already limited by the validator to 36 chars, we don't need to do anything

**Comment by lohanidamodar** [comment]
@christyjacob4 all the comments should be addressed.

File: app/controllers/api/storage.php (hunk 31 above)
Comment (christyjacob4) [reply app/controllers/api/storage.php:789]: Okay perfect 👍

File: app/http.php (hunk 28 above)
Comment (christyjacob4) [reply app/http.php:164]: okay 👍

File: app/init.php (hunk 23 above)
Comment (christyjacob4) [reply app/init.php:72]: Alright makes sense 👍
